x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractAggregationTestCase.java

```diff
@@ -50,6 +50,37 @@ protected static Iterable<Object[]> parameterSuppliersFromTypedDataWithDefaultCh
         return parameterSuppliersFromTypedData(randomizeBytesRefsOffset(suppliers));
     }
 
+    /**
+     * Converts a list of TypedDataSuppliers into a multirow TypedDataSupplier using those cases.
+     * <p>
+     *     E.g. {@code toMultiRow(1, 1000, TestCaseSupplier.dateCases())}
+     * </p>
+     * @param minRows minimum number of rows (inclusive)
+     * @param maxRows maximum number of rows (inclusive)
+     * @param values list of TypedDataSuppliers to use
+     */
+    protected static TestCaseSupplier.TypedDataSupplier toMultiRow(
+        int minRows,
+        int maxRows,
+        List<TestCaseSupplier.TypedDataSupplier> values
+    ) {
+        if (values.isEmpty()) {
+            throw new IllegalArgumentException("values cannot be empty");
+        }
+
+        var type = values.get(0).type();
+
+        return new TestCaseSupplier.TypedDataSupplier(
+            type + " rows",
+            () -> IntStream.range(0, randomIntBetween(minRows, maxRows))
+                .mapToObj(i -> values.get(randomIntBetween(1, values.size()) - 1).get().getValue())
```

Comment:
I think `randomFrom(values).get().getValue()` is the more idiomatic way to do this.